README.md

```diff
@@ -1 +1,629 @@
+│  ┌────────UCAN───────┐  │
+│  │                   │  │
+│  │  iss: Responder   │  │
+│  │  aud: ReqECDH     │  │
+│  │  att: []          │  │
+│  │  fct: nextResECDH │  │
+│  │  prf: ...         │  │
+│  │                   │  │
+│  └───────────────────┘  │
+│                         │
+└─────────────────────────┘
+```
+
+Upon receipt, the Requestor MUST validate that the UCAN capabilities fulfill their `caps` criteria. The UCAN itself MUST be valid, unrevoked, unexpired, and intended for the temporary DID (the `aud` field). If any of these checks fail, the session MUST be abandoned, the temporary DID regenerated, and the protocol restarted from [intention broadcast](#32-requestor-broadcasts-intent).
+
+### 3.3.1 Validation UCAN
+
+The validation UCAN MUST NOT be used to delegate any capabilities. This UCAN MUST only be used to prove access to capabilities and sign the AES key. The `att` and `my` fields MUST be empty arrays. The issuer (`iss`) field MUST contain the Responder's long-term DID (rather than the temporary ECDH DID). The audience (`aud`) field MUST contain the Requestor's temporary ECDH DID from [§3.2](#32-requestor-broadcasts-intent).
+
+This UCAN MUST be encrypted with the [KDF-generated AES-GCM key](#143-key-derivation) plus IV before being placed into the payload in [§3.3.2](#332-payload).
+
+#### 3.3.1.1 Challenge
+
+The Responder MUST set the method of challenge to validate the Requestor. This MUST be set in the `fct` section of the UCAN so that it is signed by the Responder. The RECOMMENDED authorization methods are out-of-band PIN validation (`oob-pin`) and UCAN (`ucan`).
+
+To set the challenge as `oob-pin`, the `fct` section of the UCAN MUST include the following:
+
+``` javascript
+{
+  ...,
+  "fct": [
+    ...,
+    {"awake/challenge": "oob-pin"}
+  ]
+}
+```
+
+To set the challenge as `ucan`, the `fct` section of the UCAN MUST include the following:
+
+``` javascript
+{
+  ...,
+  "fct": [
+    ...,
+    { 
+      "awake/challenge": "ucan",
+      "caps": [...requiredCaps] 
+    }
+  ]
+}
+```
+
+If more than one `awake/challenge` field is set, the lowest-indexed one MUST be used.
+
+#### 3.3.1.2 Next Responder ECDH
+
+The UCAN's facts (`fct`) field MUST also include the next Responder ECDH public key (to be used in Step 4) encoded as `did:key`. Having the next key in the UCAN places it inside the signature envelope, associating the next key with the Responder's UCAN DID.
+
+``` javascript
+//JSON encoded
+{
+  ...,
+  "fct": [
+    ...,
+    {"awake/nextdid": step4EcdhDid}
+  ]
+}
+```
+
+If more than one `awake/nextdid` field is set, the lowest-indexed one MUST be used.
+
+### 3.3.2 Payload
+
+To start the Double Ratchet, the payload in this stage has the highest number of cleartext fields. Note that the value in the `res` field contain the temporary ECDH DIDs, and MUST NOT use the Responder's actual long-term DID. Conversely, the UCAN inside the encrypted payload MUST use the Responder's long-term DID.
```

Comment:
> Note that the value in the `res` field contain the temporary ECDH DIDs

Should `res` be the `iss`? I see `res` in the message type, but not as a field.

Reply:
It should! Thanks :)